### PR TITLE
Use Promises generic types

### PR DESCRIPTION
**Pull request by Ndiritu**
related to https://github.com/microsoft/kiota/issues/3542

### CODE REVIEW

**Comment by Ndiritu** [comment]
Pending release of https://github.com/microsoft/kiota-abstractions-php/pull/99

**Review by SilasKenneth — APPROVED** [review]

File: src/PhpLeagueAccessTokenProvider.php [hunk 1]
```diff
@@ -124,7 +124,7 @@ public function getAuthorizationTokenAsync(string $url, array $additionalAuthent
             if (($scheme !== 'https' || !$this->getAllowedHostsValidator()->isUrlHostValid($url))
                 && !$isLocalhost) {
                 $span->setAttribute(self::URL_VALID_KEY, false);
-                return new FulfilledPromise(null);
+                throw new InvalidArgumentException("Invalid URL provided. URL must be HTTPS and part of the allowed hosts.");
```
Comment (SilasKenneth) [inline src/PhpLeagueAccessTokenProvider.php:127]: What about localhost URLs?

**Review by SilasKenneth — CHANGES_REQUESTED** [review]

File: src/PhpLeagueAccessTokenProvider.php (hunk 1 above)
Comment (Ndiritu) [reply src/PhpLeagueAccessTokenProvider.php]: Updated to align with other languages where invalid URLs don't cause an exception but an exception is thrown if the URL is not HTTPS and is a non-localhost URL.

Comment (Ndiritu) [reply src/PhpLeagueAccessTokenProvider.php]: https://github.com/microsoft/kiota-abstractions-php/pull/105

**Review by SilasKenneth — APPROVED** [review]

**Pull request merged**